### PR TITLE
Checking that the effectiveLineNumber is valid before accessing the line content

### LINKED ISSUE #199938: Error: Illegal value for lineNumber
I was switching between files, opened an empty file and got the error below

```
ERR Illegal value for lineNumber: Error: Illegal value for lineNumber
    at TextModel.getLineContent (vscode-file://vscode-app/Users/jrieken/Code/vscode/out/vs/editor/common/model/textModel.js:572:23)
    at LightBulbWidget.update (vscode-file://vscode-app/Users/jrieken/Code/vscode/out/vs/editor/contrib/codeAction/browser/lightBulbWidget.js:170:78)
    at CodeActionController.update (vscode-file://vscode-app/Users/jrieken/Code/vscode/out/vs/editor/contrib/codeAction/browser/codeActionController.js:98:42)
```

### PR DESCRIPTION
**Pull request by aiday-mar**
Fixes https://github.com/microsoft/vscode/issues/199938

### CODE REVIEW

**Comment by aeschli** [comment]
The problem is on line 205, where the effectiveLineNumber is increased without checking whether this is possible.

**Comment by aiday-mar** [comment]
Integration test errors appear to not be related at all to the change, this is quite curious, this needs some investigation

File: src/vs/editor/contrib/codeAction/browser/lightBulbWidget.ts [hunk 1]
```diff
@@ -201,7 +201,7 @@ export class LightBulbWidget extends Disposable implements IContentWidget {
 		if (!lineHasSpace) {
 			if (lineNumber > 1 && !isFolded(lineNumber - 1)) {
 				effectiveLineNumber -= 1;
-			} else if (!isFolded(lineNumber + 1)) {
+			} else if (!isFolded(lineNumber + 1) && (lineNumber + 1 <= model.getLineCount())) {
```
Comment (aeschli) [inline src/vs/editor/contrib/codeAction/browser/lightBulbWidget.ts:204]: can be written as `lineNumber < model.getLineCount()`

**Review by aeschli — DISMISSED** [review]

**Review by aeschli — DISMISSED** [review]

File: extensions/typescript-language-features/src/languageFeatures/quickFix.ts [hunk 2]
```diff
@@ -262,6 +262,8 @@ class TypeScriptQuickFixProvider implements vscode.CodeActionProvider<VsCodeCode
 		}
 
 		const allActions = Array.from(results.values);
+		console.log('allActions : ', allActions);
```
[the remark this answers is not in the document]
Comment (aiday-mar) [reply extensions/typescript-language-features/src/languageFeatures/quickFix.ts:265]: sorry, those were console logs from some other work I was looking into. It was urelated to the pr

**Comment by aiday-mar** [comment]
Yesterday during our discussion you mentioned that we could reuse the calculation in the variable `lineHasSpace` to calculate the effective column number. After some inspection I am not sure this would make the code clearer. The current effective column number is calculated as follows:

`!!model.getLineContent(effectiveLineNumber).match(/^\S\s*$/) ? 2 : 1`

So the effective column number is:
- 2 is there is only one character at the beginning of the line, and the rest are whitespace
- 1 otherwise

So the calculations are rather different

**Comment by aeschli** [comment]
ok. sure. Buyt still, the check to go on column 2 should only happen if we increased or decreased the effective line number.

**Comment by aeschli** [comment]
ok. sure. But still, the check and column update should only happen when there was no space on the original line.

**Comment by aiday-mar** [comment]
Ah yes now I understand, will do 👍🏻

**Comment by aiday-mar** [comment]
I added the code into the if statement

**Review by alexdima — APPROVED** [review]
LGTM. @justschen Please also take a look.

**Review by justschen — APPROVED** [review]
yes lgtm!

**Pull request merged**